src/middlewared/middlewared/utils/interface.py

```diff
@@ -0,0 +1,43 @@
+import contextlib
+import os
+import time
+
+
+IFACE_LINK_STATE_MAX_WAIT: int = 60
+RTF_GATEWAY: int = 0x0002
+RTF_UP: int = 0x0001
+
+
+def get_default_interface() -> str | None:
+    data = []
+    with contextlib.suppress(FileNotFoundError):
+        with open('/proc/net/route', 'r') as f:
+            data = [line.split() for line in f.readlines()][1:]
```

Comment:
There is no reason to do `data = []` and then do `readlines()[1:]`. You're still reading the entire file into memory and then iterating it again. I'd rather iterate over the file object and return early.